### PR TITLE
KAFKA-2657; Kafka clients fail to start if one of broker isn't resolved by DNS

### PR DESCRIPTION
**Pull request by apakulov**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/ClientUtils.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
                 try {
                     InetSocketAddress address = new InetSocketAddress(host, port);
                     if (address.isUnresolved())
-                        throw new ConfigException("DNS resolution failed for url in " + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG + ": " + url);
+                        log.error("DNS resolution failed for url in " + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG + ": " + url);
                     addresses.add(address);
```
Comment (lindong28) [inline clients/src/main/java/org/apache/kafka/clients/ClientUtils.java]: Do you want to add `address` to `addresses` even when `address` is not resolved?

Comment (apakulov) [inline clients/src/main/java/org/apache/kafka/clients/ClientUtils.java]: @lindong28 I've been trying to less strict, but I think you're right. I've updated my PR.

**Comment by lindong28** [comment]
Thanks for the patch. The implementation looks good to me. But I am not sure if the motivation is strong. The exception is only thrown if the client can not resolve broker's address during initialization. Note that DNS resolution failure is relatively rare and usually happen if the address is wrong. Why no just let the client fail so that operator can immediately look into the issue?

You can also ping a committer for comments.

**Comment by xstevens** [comment]
We ran into this issue today. We were doing a rolling upgrade of existing brokers, when one of our services that's a producer was rolled out as well. This prevented our producer service from starting up when the rest of the brokers were fine. It would be nice to have this patch put into both 0.8.2.x and 0.9.x.

**Comment by ijuma** [comment]
Thanks for the PR. Sorry for not getting to this sooner. The issue that the PR tries to solve was resolved via https://github.com/apache/kafka/commit/c4bbf342432291220242ad4177fd72a959ddcb94 so feel free to close this PR (we can't do it ourselves).

**Pull request closed without merging**